src/components/Email/ManageWalletModalWithEmail.tsx

```diff
@@ -0,0 +1,193 @@
+import { useCallback, useEffect, useState } from 'react';
+import { graphql, useFragment } from 'react-relay';
+import styled from 'styled-components';
+
+import breakpoints from '~/components/core/breakpoints';
+import { Button } from '~/components/core/Button/Button';
+import { HStack, VStack } from '~/components/core/Spacer/Stack';
+import { BaseM, TitleDiatypeL, TitleDiatypeM } from '~/components/core/Text/Text';
+import Toggle from '~/components/core/Toggle/Toggle';
+import EmailManager from '~/components/Email/EmailManager';
+import ManageWallets from '~/components/ManageWallets/ManageWallets';
+import { useReportError } from '~/contexts/errorReporting/ErrorReportingContext';
+import { useToastActions } from '~/contexts/toast/ToastContext';
+import { ManageWalletModalWithEmailFragment$key } from '~/generated/ManageWalletModalWithEmailFragment.graphql';
+
+import useUpdateEmailNotificationSettings from './useUpdateEmailNotificationSettings';
+
+type Props = {
+  queryRef: ManageWalletModalWithEmailFragment$key;
+  newAddress?: string;
+  onEthAddWalletSuccess?: () => void;
+  onTezosAddWalletSuccess?: () => void;
+};
+
+function ManageWalletsModalWithEmail({
+  newAddress,
+  queryRef,
+  onEthAddWalletSuccess,
+  onTezosAddWalletSuccess,
+}: Props) {
+  const query = useFragment(
+    graphql`
+      fragment ManageWalletModalWithEmailFragment on Query {
+        viewer @required(action: THROW) {
+          ... on Viewer {
+            email @required(action: THROW) {
+              email
+              emailNotificationSettings {
+                unsubscribedFromAll
+                unsubscribedFromNotifications
+              }
+            }
+          }
+        }
+
+        ...EmailManagerFragment
+        ...ManageWalletsFragment
+      }
+    `,
+    queryRef
+  );
+
+  const updateEmailNotificationSettings = useUpdateEmailNotificationSettings();
+  const [isEmailNotificationChecked, setIsEmailNotificationChecked] = useState(false);
+  const [isShowAddEmail, setIsShowAddEmail] = useState(false);
+  const { pushToast } = useToastActions();
+  const reportError = useReportError();
+
+  // Invert the value since in the db stored as unsubscribe while in the UI it's subscribe
+  const isEmailNotificationSubscribed =
+    query?.viewer?.email?.emailNotificationSettings?.unsubscribedFromNotifications || false;
+
+  const isEmailUnsubscribedFromAll =
+    !!query?.viewer?.email?.emailNotificationSettings?.unsubscribedFromAll ?? false;
```

Comment:
hmm i think this can just be the following, without the need for fallbacks, because the `!` will coerce them into booleans:

```js
const isEmailNotificationSubscribed = !query?.viewer?.email?.emailNotificationSettings?.unsubscribedFromNotifications

const isEmailUnsubscribedFromAll = !!query?.viewer?.email?.emailNotificationSettings?.unsubscribedFromAll
```

Reply:
Removed the fallback!